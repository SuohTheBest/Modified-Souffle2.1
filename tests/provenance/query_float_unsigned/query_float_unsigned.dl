// Souffle - A Datalog Compiler
// Copyright (c) 2020, The Souffle Developers. All rights reserved
// Licensed under the Universal Permissive License v 1.0 as shown at:
// - https://opensource.org/licenses/UPL
// - <souffle root>/licenses/SOUFFLE-UPL.txt

//
// query_float_unsigned.dl
// This code tests usage of float/unsigned with query command.
//

.pragma "provenance" "explain"

.decl Float(x: float)
Float(1).
Float(1.5).
Float(0).
Float(3.5).
Float(-0.5).



.decl Unsigned(x: unsigned)
Unsigned(0).
Unsigned(1).
Unsigned(0b10).
Unsigned(0xf).
Unsigned(0x11).


.output Unsigned, Float
