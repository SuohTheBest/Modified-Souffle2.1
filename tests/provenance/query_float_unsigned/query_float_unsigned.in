query Float(x)
;
;
;
;
query Unsigned(x)
;
;
;
;
exit
