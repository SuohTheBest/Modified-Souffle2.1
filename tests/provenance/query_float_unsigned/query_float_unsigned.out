x = -0.5
x = 0
x = 1
x = 1.5
x = 3.5.
x = 0
x = 1
x = 2
x = 15
x = 17.
